### PR TITLE
fix: Add button role to table header interactive element

### PR DESCRIPTION
**Pull request by StoianLucian**
## Description

[SITES-10962](https://jira.corp.adobe.com/browse/SITES-10962)

Button: Button does not have a role.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: coral-component-table/src/scripts/Table.js [hunk 1]
```diff
@@ -2460,6 +2460,12 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
       if (this.head && this.head.sticky) {
         headerCell.setAttribute('role', 'presentation');
         headerCell._elements.content.setAttribute('role', 'columnheader');
+        if(!headerCell.children.length > 0) {
+          headerCell._elements.content.setAttribute('role', 'button');
+          if(!headerCell.firstChild) {
+            headerCell._elements.content.setAttribute('role', 'columnheader');
```
Comment (Pareesh) [inline coral-component-table/src/scripts/Table.js:2466]: Is this change required only when columnheader is sticky.

Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js]: The change is not required only when the columnheader is sticky, I moved the functionality so it triggers regardless of the sticky state

File: coral-component-table/src/scripts/Table.js [hunk 2]
```diff
@@ -2467,6 +2467,12 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
       headerCell.setAttribute('role', 'rowheader');
     }
     headerCell.setAttribute('scope', scope);
+    if(!headerCell.children.length > 0) {
+      headerCell._elements.content.setAttribute('role', 'button');
+      if(!headerCell.firstChild) {
+        headerCell._elements.content.setAttribute('role', 'columnheader');
+      }
```
Comment (Pareesh) [inline coral-component-table/src/scripts/Table.js:2474]: headercell will be a columnheader, why are setting role to button ??

Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js:2474]: The ticket requires that the interactive elements of the headercell must have a button role 
SITES-10962

Comment (Pareesh) [reply coral-component-table/src/scripts/Table.js:2474]: Hi @StoianLucian ,
The ticket suggestion is not always the best and feasible solution and might not be aware of the component usage.  The exact problem is the headercell doesn't have a role, you need to add the most apt role to it.

Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js:2474]: Hi @Pareesh Since the interactive elements act as a sorting button for the content below (**SITES-10962.png**) won`t the role 'button" be the most appropriate one?
SITES-10962

Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js:2474]: Hi @Pareesh I changed the logic to apply the button role only if the parent element (wrapper) has the sortable attribute. It doesn`t change the role of the non-interactive rows

Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js]: HI @Pareesh I had to move the functionality in order to apply the changes since the sortable attribute was applied later in the original function

File: coral-component-table/src/scripts/Table.js [hunk 3]
```diff
@@ -1598,6 +1598,10 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
       // For icons (chevron up/down) styling
       headerCell[column.sortable ? 'setAttribute' : 'removeAttribute']('sortable', '');
 
+      if(headerCell.hasAttribute('sortable')){
+        headerCell.firstChild.setAttribute('role', 'button');
```
Comment (Pareesh) [inline coral-component-table/src/scripts/Table.js:1602]: please use same changes as done for coralui3

**Comment by Pareesh** [comment]
Please check the failing build as well.

File: coral-component-table/src/scripts/Table.js (hunk 3 above)
Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js]: @Pareesh I modified so the changes are the same as for coralui3. Also the build seems to fail because of a test that`s not related to my change "#**coral-columnview:navigate
        ✖ should be triggered when a column is removed**"

**Comment by ens84646** [comment]
@Pareesh, failing test not related to the actual change. Can you please help us with approving/merging this PR?

**Review by petcud — APPROVED** [review]

**Review by cbogdan0707 — APPROVED** [review]

File: coral-component-table/src/scripts/Table.js [hunk 4]
```diff
@@ -2467,6 +2467,10 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
       headerCell.setAttribute('role', 'rowheader');
     }
     headerCell.setAttribute('scope', scope);
+      
+    if(headerCell.hasAttribute('sortable')){
+      headerCell._elements.content.setAttribute('role', 'button');
```
Comment (Pareesh) [inline coral-component-table/src/scripts/Table.js:2472]: I think `headerCell.content` also points to `headerCell._elements.content`. Please change to `headerCell.content`

Comment (StoianLucian) [reply coral-component-table/src/scripts/Table.js]: Hi @Pareesh I made the change

**Review by Pareesh — APPROVED** [review]
